analysers/analyser_osmosis_highway_vs_building.py

```diff
@@ -205,6 +233,32 @@
         ST_Intersects(ST_Buffer(power.geom::geography, 0.25)::geometry, highway.linestring)
 """
 
+sql32 = """
+SELECT
+    DISTINCT ON(commercial.id)
+    commercial.id,
+    highway.id,
+    ST_AsText(commercial.geom)
+FROM
+    {0}commercial AS commercial
+    JOIN {1}highway AS highway ON
+        highway.level = '0' AND
+        commercial.layer = highway.layer AND
+        commercial.buffergeom && highway.linestring AND
+        ST_Intersects(commercial.buffergeom, highway.linestring) AND
+        (
+            -- exclude shops nearly touching service roads (e.g. unmapped drivethroughs), pedestrian ways (e.g. shopping streets), ...
+            highway.highway IS NULL OR -- railways are also in table highway
+            highway.highway IN ('living_street', 'residential', 'unclassified', 'tertiary', 'tertiary_link',
+                              'secondary', 'secondary_link', 'primary', 'primary_link', 'trunk', 'trunk_link',
+                              'motorway', 'motorway_link', 'busway', 'bus_guideway')
+        )
+ORDER BY
+    commercial.id,
+    highway.id,
+    commercial.geom
```

Comment:
```suggestion
    highway.id
```

Reply:
Thanks! I keep forgetting this with ST_AsText 🙈